packages/mermaid/src/rendering-util/handle-markdown-text.spec.ts

```diff
@@ -0,0 +1,388 @@
+        { content: 'test', type: 'normal' },
+      ],
+    ];
+
+    const output = markdownToLines(input);
+    expect(output).toEqual(expectedOutput);
+  });
+
+  test('paragraph 1', () => {
+    const input = `**Start** with
+    a second line`;
+
+    const expectedOutput = [
+      [
+        { content: 'Start', type: 'strong' },
+        { content: 'with', type: 'normal' },
+      ],
+      [
+        { content: 'a', type: 'normal' },
+        { content: 'second', type: 'normal' },
+        { content: 'line', type: 'normal' },
+      ],
+    ];
+
+    const output = markdownToLines(input);
+    expect(output).toEqual(expectedOutput);
+  });
+
+  test('paragraph', () => {
+    const input = `**Start** with
+
+    a second line`;
+
+    const expectedOutput = [
+      [
+        { content: 'Start', type: 'strong' },
+        { content: 'with', type: 'normal' },
+      ],
+      [
+        { content: 'a', type: 'normal' },
+        { content: 'second', type: 'normal' },
+        { content: 'line', type: 'normal' },
+      ],
+    ];
+
+    const output = markdownToLines(input);
+    expect(output).toEqual(expectedOutput);
+  });
+
+  test('Only italic formatting', () => {
+    const input = `This is an *italic* test`;
+    const output = markdownToLines(input);
+    expect(output).toMatchInlineSnapshot(`
+    [
+      [
+        {
+          "content": "This",
+          "type": "normal",
+        },
+        {
+          "content": "is",
+          "type": "normal",
+        },
+        {
+          "content": "an",
+          "type": "normal",
+        },
+        {
+          "content": "italic",
+          "type": "emphasis",
+        },
+        {
+          "content": "test",
+          "type": "normal",
+        },
+      ],
+    ]
+  `);
+  });
+
+  test('Mixed formatting', () => {
+    const input = `*Italic* and **bold** formatting`;
+    const output = markdownToLines(input);
+    expect(output).toMatchInlineSnapshot(`
+    [
+      [
+        {
+          "content": "Italic",
+          "type": "emphasis",
+        },
+        {
+          "content": "and",
+          "type": "normal",
+        },
+        {
+          "content": "bold",
+          "type": "strong",
+        },
+        {
+          "content": "formatting",
+          "type": "normal",
+        },
+      ],
+    ]
+  `);
+  });
+
+  test('Mixed formatting with newline', () => {
+    const input = `The dog in **the** hog... a *very long text* about it
+Word!`;
+
+    const output = markdownToLines(input);
+    expect(output).toMatchInlineSnapshot(`
+    [
+      [
+        {
+          "content": "The",
+          "type": "normal",
+        },
+        {
+          "content": "dog",
+          "type": "normal",
+        },
+        {
+          "content": "in",
+          "type": "normal",
+        },
+        {
+          "content": "the",
+          "type": "strong",
+        },
+        {
+          "content": "hog...",
+          "type": "normal",
```

Comment:
Micromark is parsing special characters properly. 

So I've removed this handling. 
<img width="786" alt="image" src="https://user-images.githubusercontent.com/10703445/234177907-4befa3a8-8412-4594-ab35-c0026b375e97.png">